### ISSUE TITLE
WooCommerce - referrals with empty status values

### ISSUE THREAD

**Issue by gspice** [AffiliateWP/AffiliateWP#2072]
I saw a closed issue on this here:
https://github.com/AffiliateWP/AffiliateWP/issues/1042

This is the current ticket, I've asked him for admin access in case we need it:
https://secure.helpscout.net/conversation/327234520/54296/?folderId=205529

He's attached the referrals table exported if that helps:

[wpbs_affiliate_wp_referrals_3-2-2017.csv.zip](https://github.com/AffiliateWP/AffiliateWP/files/817714/wpbs_affiliate_wp_referrals_3-2-2017.csv.zip)

Here is the screen shot:

![affiliatewp_pending_referrals](https://cloud.githubusercontent.com/assets/3018245/23566312/2c3ba3d8-0006-11e7-929f-81981811ffc8.png)

**Comment by pippinsplugins** [comment]
@gspice when he comes back with admin access, try going to the referral edit screen, selecting `Pending` from the status drop down, then updating it. Does that resolve it?

**Comment by gspice** [comment]
@pippinsplugins Yep, I edited the two referral individually and set to a Pending status and it "stuck", meaning he can now click to Accept. I will close this issue.

**Comment by gspice** [comment]
This user responded again today and has more WooCommerce referrals with a blank status entries.  He can manually edit but the question is why do they keep appearing?    Payment shows PayPal - IPN and PDT entries on both of them, not sure if that helps:

The login credentials are on the ticket:

https://secure.helpscout.net/conversation/327234520/54296/?folderId=1170842

**Comment by gspice** [comment]
Yay!  2.0.5 - I will ask the user if he can leave those blank status entries as-is for a little bit.

**Comment by ramiabraham** [comment]
@AffiliateWP/core I'm unable to replicate this locally. Awaiting admin credentials on customer site. My hunch is that an add-on is the cause. Can anyone replicate?

**Comment by pippinsplugins** [comment]
Happened again https://secure.helpscout.net/conversation/332043657/54905/?folderId=205530

**Comment by pippinsplugins** [comment]
I'm thinking this has to do with the order failing and then being completed. See these order notes:
<img width="312" alt="screen shot 2017-03-31 at 3 46 50 pm" src="https://cloud.githubusercontent.com/assets/1034109/24568874/5ba23d42-1629-11e7-8836-1bc3cce416a4.png">

**Comment by gspice** [comment]
Have another ticket with empty status values, and the Woo orders I viewed didn't fail.  Admin access is on the ticket:

https://secure.helpscout.net/conversation/343635258/56648/?folderId=1170842

Also on this one, we he tries to edit the status of the referral , any status of any referral, he receives these PHP errors referencing our AffiliateWP Store Credit plugin. I deleted the plugin and reloaded just in case there was an issue with it but the errors remain.  Let me know if this should be a separate issue on it's on, I wasn't sure:

`Warning: Missing argument 2 for AffiliateWP_Store_Credit_Base::process_payout() in /home/kronicju/public_html/wp-content/plugins/affiliatewp-store-credit/integrations/class-base.php on line 80

Warning: Missing argument 3 for AffiliateWP_Store_Credit_Base::process_payout() in /home/kronicju/public_html/wp-content/plugins/affiliatewp-store-credit/integrations/class-base.php on line 80

Warning: Cannot modify header information - headers already sent by (output started at /home/kronicju/public_html/wp-content/plugins/affiliatewp-store-credit/integrations/class-base.php:80) in /home/kronicju/public_html/wp-includes/pluggable.php on line 1179`

**Comment by sumobi** [comment]
Note, the store credit errors above were recently fixed in https://github.com/AffiliateWP/affiliatewp-store-credit/issues/40

I've also replicated this issue on a customer's site without store credit so we can at least rule that out. Going to do some digging and see if I can work out the exact user flow that causes the pending status without icon.

**Comment by sumobi** [comment]
Ok here's how to replicate it (tested on customer's site and also locally):

1. Add product to cart and proceed to checkout
2. Select PayPal as payment option and click purchase but do not follow through with PayPal purchase (stay on PayPal screen)
3. In the admin a "pending" referral will be created as expected
4. Back on the PayPal page, click and hold the browser's back button and select the checkout screen (clicking back will probably work also).
5. Select the PayPal option again (it will be unselected) and click to purchase again
6. Upon landing on PayPal again, refresh the referrals screen in the admin and the status will show without an icon.

I suspect this is happening because customer's land on PayPal but want to double check they entered the correct details. So they go back to checkout, everything looks good and then they purchase again. However once they get to PayPal they abandon their order without cancelling and don't come back.

@AffiliateWP/core Can anyone else replicate following the steps above? 

Edit: Hah... just looked at the older ticket which was resolved and it actually outlines how to replicate the issue. I now remember testing, replicating, and seeing the issue resolved. Guess we reintroduced the error? https://github.com/AffiliateWP/AffiliateWP/issues/1042#issuecomment-168103230

Or am I not understanding what this issue is? Above it mentions "Blank status entries" (https://github.com/AffiliateWP/AffiliateWP/issues/2072#issuecomment-286247587)
